InjectionIII/InjectionServer.mm

```diff
@@ -180,23 +180,35 @@ - (void)runInBackground {
     [self setProject:projectFile];
 
     // read status requests from client app
-    while (NSString *response = [self readString])
-        if ([response hasPrefix:@"COMPLETE"])
+    InjectionCommand command;
+    while ((command = (InjectionCommand)[self readInt]) != InjectionEOF) {
+        switch (command) {
+        case InjectionComplete:
             [appDelegate setMenuIcon:@"InjectionOK"];
-        else if ([response hasPrefix:@"PAUSE "])
+            break;
+        case InjectionPause:
             pause = [NSDate timeIntervalSinceReferenceDate] +
-                [response substringFromIndex:@"PAUSE ".length].doubleValue;
-        else if ([response hasPrefix:@"SIGN "])
-            [self writeString:[SignerService codesignDylib:[response
-                substringFromIndex:@"SIGN ".length]] ? @"SIGNED 1" : @"SIGNED 0"];
-        else if ([response hasPrefix:@"ERROR "])
+                [self readString].doubleValue;
+            break;
+        case InjectionSign: {
+            BOOL signedOK = [SignerService codesignDylib:[self readString]];
+            [self writeCommand:InjectionSigned withString: signedOK ? @"1": @"0"];
+            break;
+        }
+        case InjectionError:
             [appDelegate setMenuIcon:@"InjectionError"];
+            NSLog(@"Injection error: %@", [self readString]);
 //            dispatch_async(dispatch_get_main_queue(), ^{
 //                [[NSAlert alertWithMessageText:@"Injection Error"
 //                                 defaultButton:@"OK" alternateButton:nil otherButton:nil
 //                     informativeTextWithFormat:@"%@",
 //                  [dylib substringFromIndex:@"ERROR ".length]] runModal];
 //            });
+            break;
+        default:
+            break;
```

Comment:
Should we log here as we do in the `InjectionClient.m,`? There are no code paths leading to this case but still, just a thought.